### PR TITLE
Add missing flags for materials

### LINKED ISSUE #2021: Missing recipe for Diamond Screwdriver (LV)
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

1.4.1+patch.2

### Minecraft Version

1.20.1

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

Unsure

### Other Installed Mods

Modpack: GregTech Community Pack Modern

### Expected Behavior

The Diamond Screwdriver (LV) has a recipe.

### Actual Behavior

The Diamond Screwdriver (LV) does not have a recipe.

### Steps to Reproduce

Left click on the Diamond Screwdriver (LV) in JEI.

### Additional Information

JEI also does not find any item called "Diamond Screwdriver Tip" that would be required for this recipe.

### PR DESCRIPTION
**Pull request by JuiceyBeans**
## What
- Adds missing flags to a ton of materials so that their tool recipes generate
- Adds debug lines to catch missing flags for materials
- Adds styrene butadiene and silicone rubber to `ToolRecipeHandler#softMaterials`

## Implementation
`softMaterials` array has been moved from `ToolRecipeHandler#addWoodRecipes`

## Outcome
A lot of tool parts are now craftable
Fixes #2021

### CODE REVIEW

**Review by screret — APPROVED** [review]

**Comment by JuiceyBeans** [comment]
Before merging, do check whether the debug lines should be left in. It doesn't find anymore missing flags and could be useful for future materials

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 1]
```diff
@@ -327,8 +345,14 @@ private static void processElectricTool(TagPrefix prefix, Material material, Too
                             .duration((int) material.getMass() * 4)
                             .EUt(8L * voltageMultiplier)
                             .save(provider);
+                } else {
+                    GTCEu.LOGGER.info("Did not find gear for for " + material.getName() +
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 2]
```diff
@@ -239,6 +254,9 @@ private static void processTool(TagPrefix prefix, Material material, ToolPropert
             addDyeableToolRecipe(provider, material, GTToolType.CROWBAR, true,
                     "hDS", "DSD", "SDf",
                     'S', rod);
+        } else if (!ArrayUtils.contains(softMaterials, material)) {
+            GTCEu.LOGGER.info("Did not find rod for for " + material.getName() +
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 3]
```diff
@@ -228,7 +237,13 @@ private static void processTool(TagPrefix prefix, Material material, ToolPropert
                             'P', plate,
                             'T', new UnificationEntry(TagPrefix.screw, material),
                             'S', rod);
+                } else if (!ArrayUtils.contains(softMaterials, material)) {
+                    GTCEu.LOGGER
+                            .info("Did not find bolt for for " + material.getName() + ", skipping wirecutter recipe");
                 }
+            } else {
+                GTCEu.LOGGER.info("Did not find plate for for " + material.getName() +
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 4]
```diff
@@ -228,7 +237,13 @@ private static void processTool(TagPrefix prefix, Material material, ToolPropert
                             'P', plate,
                             'T', new UnificationEntry(TagPrefix.screw, material),
                             'S', rod);
+                } else if (!ArrayUtils.contains(softMaterials, material)) {
+                    GTCEu.LOGGER
+                            .info("Did not find bolt for for " + material.getName() + ", skipping wirecutter recipe");
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 5]
```diff
@@ -327,8 +345,14 @@ private static void processElectricTool(TagPrefix prefix, Material material, Too
                             .duration((int) material.getMass() * 4)
                             .EUt(8L * voltageMultiplier)
                             .save(provider);
+                } else {
+                    GTCEu.LOGGER.info("Did not find gear for for " + material.getName() +
+                            ", skipping gear -> buzzsaw blade recipe");
                 }
             }
+        } else {
+            GTCEu.LOGGER.info("Did not find plate for for " + material.getName() +
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

File: src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java [hunk 6]
```diff
@@ -341,6 +365,9 @@ private static void processElectricTool(TagPrefix prefix, Material material, Too
                         ChemicalHelper.get(toolPrefix, material),
                         "fR", " h",
                         'R', new UnificationEntry(TagPrefix.rodLong, material));
+            } else {
+                GTCEu.LOGGER.info("Did not find long rod for for " + material.getName() +
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/recipe/generated/ToolRecipeHandler.java]: for for

**Review by screret — CHANGES_REQUESTED** [review]

**Review by screret — APPROVED** [review]

**Pull request merged**